.github/workflows/flatpak.yml

```diff
@@ -19,35 +24,91 @@ env:
-      if: github.event_name == 'pull_request'
-      shell: bash
-      run: |
-        LABELS_URL="$(echo ${{ github.event.pull_request.url }} | sed s'/pulls/issues/')"
-        LABEL_FOUND="$(curl -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "${LABELS_URL}/labels" | sed -n 's/.*"name": "\(.*\)",/\1/p' | grep 'Seeking Testers' || true)"
-        if [ "${LABEL_FOUND}" = "Seeking Testers" ]; then
-          echo "SEEKING_TESTERS=1" >> $GITHUB_ENV
-        else
-          echo "SEEKING_TESTERS=0" >> $GITHUB_ENV
-        fi
-
-    - name: Checkout
-      uses: actions/checkout@v2.3.3
-      if: success() && (github.event_name != 'pull_request' || env.SEEKING_TESTERS == '1')
-      with:
-        submodules: 'recursive'
-
-    - name: Build Flatpak Manifest
-      uses: bilelmoussaoui/flatpak-github-actions/flatpak-builder@v3
-      if: success() && (github.event_name != 'pull_request' || env.SEEKING_TESTERS == '1')
-      with:
-        bundle: obs-studio-${{ github.sha }}.flatpak
-        manifest-path: CI/flatpak/com.obsproject.Studio.json
-        cache-key: flatpak-builder-${{ github.sha }}
+      - name: Checkout
+        uses: actions/checkout@v2.3.3
+        with:
+          submodules: 'recursive'
+
+      - name: Build Flatpak Manifest
+        uses: bilelmoussaoui/flatpak-github-actions/flatpak-builder@master
+        with:
+          bundle: obs-studio-${{ github.sha }}.flatpak
+          manifest-path: CI/flatpak/com.obsproject.Studio.json
+          cache-key: flatpak-builder-${{ github.sha }}
+          mirror-screenshots-url: https://dl.flathub.org/repo/screenshots
+
+      - name: Validate AppStream
+        shell: bash
+        run: |
+          cd ${{ env.FLATPAK_BUILD_FOLDER }}
+          appstream-util validate appdata/com.obsproject.Studio.appdata.xml
+
+      - name: Verify icon and metadata in app-info
+        shell: bash
+        run: |
+          cd ${{ env.FLATPAK_BUILD_FOLDER }}
+          test -f app-info/icons/flatpak/128x128/com.obsproject.Studio.png || { echo "Missing 128x128 icon in app-info" ; exit 1; }
+          test -f app-info/xmls/com.obsproject.Studio.xml.gz || { echo "Missing com.obsproject.Studio.xml.gz in app-info" ; exit 1; }
+
+      - name: Commit screenshots to the OSTree repository
+        run: |
+          ostree commit --repo=repo --canonical-permissions --branch=screenshots/x86_64 flatpak_app/screenshots
+
+      - name: "Publish to Flathub Beta"
+        uses: bilelmoussaoui/flatpak-github-actions/flat-manager@v3
+        with:
+          flat-manager-url: https://hub.flathub.org/
+          repository: beta
+          token: ${{ secrets.FLATHUB_BETA_TOKEN }}
+
+      - name: "Publish to Flathub"
+        uses: bilelmoussaoui/flatpak-github-actions/flat-manager@v3
+        if: "!contains(github.ref, '-rc')"
+        with:
+          flat-manager-url: https://hub.flathub.org/
+          repository: stable
+          token: ${{ secrets.FLATHUB_TOKEN }}
```

Comment:
I assume you already have this token? How is write-access to the Flathub listing shared with others from the OBS team?

Reply:
We're in the process of generating these tokens, I'll coordinate after they're create

Reply:
Thanks to @DDRBoxman, the tokens were added to this repository!